posthog/api/action.py

```diff
@@ -104,17 +90,30 @@ def validate(self, attrs):
 
         return attrs
 
-    def create(self, validated_data: Any) -> Any:
-        steps = validated_data.pop("steps", [])
-        validated_data["created_by"] = self.context["request"].user
-        instance = super().create(validated_data)
+    def _sync_steps(self, action: Action, steps: list[dict[str, Any]]) -> None:
+        # And then also save the old model for backwards compatibility
+
+        if action.pk:
+            # This is slightly< inefficient but simple and temporary
```

Comment:
Would be useful to add a "remove after" date, unless there's already a PR ready that removes the temporary thing

Reply:
Yep this is part 1 of 3 stacked PRs